### DISCUSSION TITLE
How do I get the class name of a prediction during inference?

### DISCUSSION THREAD

**Discussion by SilverStarCoder** [open-edge-platform/anomalib#1951] [Q&A] [answered]
The below code is straigh-forward in determining the score and label (True or False, i.e Normal or Abnormal), but how do I determine what class that image is? Example, if the image was a class "bottle" or a "hazelnut", and if it was indeed abnormal, if the abnormality was "broken" or "cut", etc...

```
inferencer = OpenVINOInferencer(
    path=openvino_model_path,  # Path to the OpenVINO IR model.
    metadata=metadata,  # Path to the metadata file.
    device="CPU",  # We would like to run it on an Intel CPU.
)

predictions = inferencer.predict(image=image_path)

print(predictions.pred_score, predictions.pred_label)
```

**Comment by abc-125** [comment, marked as answer]
> how do I determine what class that image is?

You train a model for every class separately (unless it is zero-shot WinCLIP), so it can be done only manually by using the `path` variable in the `OpenVINOInferencer` in your code, or in the config file. Hopefully that makes sense.

> if the abnormality was "broken" or "cut", etc

As far as I know, existing anomaly detection models don't have such ability, they calculate just a score, which shows how normal/abnormal the image is.

**Reply by samet-akcay** [reply]
It is currently not possible to that with the existing anomalib models. I've seen some recent papers focusing on detecting the type of anomalies, but we need to check their performance before adding into anomalib. 

For example, I've recently seen this paper 

Toward Multi-class Anomaly Detection: Exploring Class-aware Unified Model against Inter-class Interference
https://arxiv.org/abs/2403.14213
